File: app/src/main/java/org/digitalcampus/oppia/adapter/TagsAdapter.java
package org.digitalcampus.oppia.adapter;


import android.content.Context;
import android.graphics.Typeface;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import org.digitalcampus.mobile.learning.R;
import org.digitalcampus.mobile.learning.databinding.RowTagBinding;
import org.digitalcampus.oppia.model.Tag;

import java.util.List;

public class TagsAdapter extends RecyclerView.Adapter<TagsAdapter.TagsViewHolder> {


    private List<Tag> tags;
    private Context context;
    private OnItemClickListener itemClickListener;


    public TagsAdapter(Context context, List<Tag> tags) {
        this.context = context;
        this.tags = tags;
    }

    @Override
    public TagsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View contactView = LayoutInflater.from(context).inflate(R.layout.row_tag, parent, false);

        // Return a new holder instance
        return new TagsViewHolder(contactView);
    }


    @Override
    public void onBindViewHolder(final TagsViewHolder viewHolder, final int position) {

        final Tag t = getItemAtPosition(position);

        viewHolder.binding.tagName.setText(t.getName());
        viewHolder.binding.tagCount.setText(String.valueOf(t.getCount()));
        if(t.isHighlight()){
            viewHolder.binding.tagName.setTypeface(null, Typeface.BOLD);
        } else {
            viewHolder.binding.tagName.setTypeface(null, Typeface.NORMAL);
        }
        if(t.getDescription() != null && !t.getDescription().trim().equals("") ){
            viewHolder.binding.tagDescription.setText(t.getDescription());
            viewHolder.binding.tagDescription.setVisibility(View.VISIBLE);
        } else {
            viewHolder.binding.tagDescription.setVisibility(View.GONE);
        }
    }

    @Override
    public int getItemCount() {
        return tags.size();
    }

    public Tag getItemAtPosition(int position) {
        return tags.get(position);
    }


    public class TagsViewHolder extends RecyclerView.ViewHolder {

        private final RowTagBinding binding;

        public TagsViewHolder(View itemView) {

            super(itemView);
            binding = RowTagBinding.bind(itemView);

            itemView.setOnClickListener(v -> {
                if (itemClickListener != null) {
                    itemClickListener.onItemClick(v, getAdapterPosition());
                }
            });
        }

    }


    public void setOnItemClickListener(OnItemClickListener listener) {
        this.itemClickListener = listener;
    }

    public interface OnItemClickListener {
        void onItemClick(View view, int position);
    }
}
 

File: app/src/main/java/org/digitalcampus/oppia/adapter/TagsAdapter.kt
package org.digitalcampus.oppia.adapter

import android.content.Context
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.digitalcampus.mobile.learning.R
import org.digitalcampus.mobile.learning.databinding.RowTagBinding
import org.digitalcampus.oppia.adapter.TagsAdapter.TagsViewHolder
import org.digitalcampus.oppia.listener.OnItemClickListener
import org.digitalcampus.oppia.model.Tag

class TagsAdapter(private val context: Context, private val tags: List<Tag>) :
    RecyclerView.Adapter<TagsViewHolder>() {

    private var itemClickListener: OnItemClickListener? = null

    inner class TagsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val binding: RowTagBinding = RowTagBinding.bind(itemView)

        init {
            itemView.setOnClickListener {
                itemClickListener?.onItemClick(itemView, adapterPosition)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TagsViewHolder {
        val contactView = LayoutInflater.from(context).inflate(R.layout.row_tag, parent, false)
        return TagsViewHolder(contactView)
    }

    override fun onBindViewHolder(viewHolder: TagsViewHolder, position: Int) {
        val t = getItemAtPosition(position)
        viewHolder.binding.tagName.text = t.name
        viewHolder.binding.tagCount.text = t.countAvailable.toString()
        if (t.isHighlight) {
            viewHolder.binding.tagName.setTypeface(null, Typeface.BOLD)
        } else {
            viewHolder.binding.tagName.setTypeface(null, Typeface.NORMAL)
        }
        if (!t.description.isNullOrBlank()) {
            viewHolder.binding.tagDescription.text = t.description
            viewHolder.binding.tagDescription.visibility = View.VISIBLE
        } else {
            viewHolder.binding.tagDescription.visibility = View.GONE
        }
    }

    override fun getItemCount(): Int {
        return tags.size
    }

    fun getItemAtPosition(position: Int): Tag {
        return tags[position]
    }

    fun setOnItemClickListener(listener: OnItemClickListener) {
        itemClickListener = listener
    }
}